File: packages/android/src/main/java/com/kozmos/components/Backdrop/Backdrop.kt
package com.kozmos.components.backdrop

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.interaction.MutableInteractionSource
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import com.kozmos.tokens.KozmosThemeTokens

@Composable
fun KozmosBackdrop(
    visible: Boolean = true,
    onTap: () -> Unit = {},
    modifier: Modifier = Modifier
) {
    if (visible) {
        Box(
            modifier = modifier
                .fillMaxSize()
                // The scrim role, as React, Figma, the dialog and the drawer
                // draw it: black at half, in both themes. It was background/800
                // at 40 % until 2026-09-22, which turns light in dark mode.
                .background(KozmosThemeTokens.semanticsOverlayScrim)
                .clickable(
                    interactionSource = remember { MutableInteractionSource() },
                    indication = null,
                    onClick = onTap
                )
        )
    }
}

File: packages/ios/Sources/Components/Backdrop/Backdrop.swift
import SwiftUI

public struct KozmosBackdrop: View {
    public var visible: Bool
    public var onTap: () -> Void
    
    public init(visible: Bool = true, onTap: @escaping () -> Void = {}) {
        self.visible = visible
        self.onTap = onTap
    }
    
    public var body: some View {
        if visible {
            // The scrim role, as React, Figma, the dialog and the drawer draw
            // it: black at half, in both themes. It was background/900 at 40 %
            // until 2026-09-22, which turns light in dark mode and lifted the
            // page instead of dimming it.
            KozmosColors.semanticsOverlayScrim
                .edgesIgnoringSafeArea(.all)
                .onTapGesture {
                    onTap()
                }
        }
    }
}
